### PR TITLE
[GLUTEN-2193][CH] Support functions array_distinct/array_union

### PR DESCRIPTION
**Pull request by exmy**
## What changes were proposed in this pull request?
There is a major difference between Spark array_distinct and CH arrayDistinct:
Spark array_distinct can remain null value in the array but CH arrayDistinct will skip it.
e.g. `select arrayDistinct([1, null, 2, null, 3, 3])` Spark return `[1, null, 2, 3]`, CH return `[1, 2, 3]`.
We should make CH arrayDistinct align with Spark.

## How was this patch tested?

UT

### CODE REVIEW

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -182,6 +182,8 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude("Map Concat")
     // Random.
     .exclude("Shuffle")
+    // TODO: ArrayDistinct should handle duplicated Double.NaN
+    .excludeByPrefix("SPARK-36741")
```
Comment (exmy) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:186]: @rui-mo Please note that velox array_distinct function is unable to handle duplicated NaN.
`- SPARK-36741: ArrayDistinct should handle duplicated Double.NaN and Float.Nan *** FAILED ***
  Incorrect evaluation: array_distinct([NaN,NaN,null,null,1.0,1.0]), actual: WrappedArray(NaN, NaN, null, 1.0), expected: List(NaN, null, 1.0) (GlutenTestsTrait.scala:263)`

Comment (rui-mo) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:186]: Got it, thank you.

**Comment by baibaichen** [comment]
没看到 spark 33 的 UT

**Review by baibaichen — APPROVED** [review]
LGTM

**Pull request merged**